src/vs/workbench/browser/parts/editor/editorActions.ts

```diff
@@ -1258,39 +1263,67 @@ export class OpenLastEditorInGroup extends AbstractNavigateEditorAction {
 	}
 }
 
-export class NavigateForwardAction extends Action {
+export class NavigateForwardAction extends Action2 {
 
 	static readonly ID = 'workbench.action.navigateForward';
 	static readonly LABEL = localize('navigateForward', "Go Forward");
 
-	constructor(
-		id: string,
-		label: string,
-		@IHistoryService private readonly historyService: IHistoryService
-	) {
-		super(id, label);
+	constructor() {
+		super({
+			id: NavigateForwardAction.ID,
+			title: { value: localize('navigateForward', "Go Forward"), original: 'Go Forward', mnemonicTitle: localize({ key: 'miForward', comment: ['&& denotes a mnemonic'] }, "&&Forward") },
+			f1: true,
+			icon: Codicon.arrowRight,
+			precondition: ContextKeyExpr.has('canNavigateForward'),
+			keybinding: {
+				weight: KeybindingWeight.WorkbenchContrib,
+				win: { primary: KeyMod.Alt | KeyCode.RightArrow },
+				mac: { primary: KeyMod.WinCtrl | KeyMod.Shift | KeyCode.Minus },
+				linux: { primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.Minus }
+			},
+			menu: [
+				{ id: MenuId.MenubarGoMenu, group: '1_history_nav', order: 2 },
+				{ id: MenuId.CommandCenter, order: 2 }
+			]
+		});
 	}
 
-	override async run(): Promise<void> {
-		await this.historyService.goForward(GoFilter.NONE);
+	run(accessor: ServicesAccessor): void {
+		const historyService = accessor.get(IHistoryService);
+
+		historyService.goForward(GoFilter.NONE);
```

Comment:
Unsure if it matters but the `goForward/Back` calls aren't awaited anymore

Reply:
@jrieken true, fixed 👍